tests/Tests.ServiceProvider/CustomAutoDataAttributes.cs

```diff
@@ -0,0 +1,39 @@
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Tests;
+
+[AttributeUsage(AttributeTargets.Method)]
+public class CustomAutoDataAttribute : AutoDataAttribute
```

Comment:
When you have multiple of these it's pretty hard to name them when you keep the names generic. Lately I try to be more specific when I name the attributes. I try to name them by the kind of data they generate or by the feature they help test, e.g. `PersistenceData`, `ServiceCollectionData`.

Reply:
Do you often have more than one in your projects?

Reply:
Yes even in smaller services/applications there's usually a few attributes each customized for specific needs. Including every customization into the same attribute can significantly decrease the execution time of the test suite. It won't be visible in the Test Explorer since the test runner reports the duration of the test only on what's inside the test method not the duration of the setup.